[Adrian Leonardo Liang] iP

AgentHagu opened the pull request:
# Neuro
> "The 2020 Dodge Charger is a four door " — [Neuro-sama](https://www.youtube.com/@Neurosama)

Introducing Neuro! A CLI chatbot that provides a multitude of tools to ease your mind! It's
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use :zap: :zap: :zap:

All you need to do is,
1. download it from [here](https://github.com/AgentHagu/ip/releases/tag/A-Jar).
2. run `java -jar neuro.jar` (as of now, double clicking does not work 😢).
3. add your tasks.
4. let it manage your tasks for you (auto-save feature 🤯)

And it is **FREE**!

Features:
- [x] Add tasks (Todos, deadlines, events)
- [x] Attach dates to tasks
- [x] Mark (or unmark) tasks as done 
- [ ] Managing deadlines (coming soon?)
- [ ] Reminders (~~coming soon~~ see my mood)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public class Neuro {
    // Some other helper methods...

    public static void main(String[] args) {
        new Neuro("data/Neuro.txt").run();
    }
}
```

The character of "Neuro" in this project is inspired by the AI streamer [Neuro-sama](https://www.youtube.com/@Neurosama).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

AgentHagu committed:
Rename Duke to Neuro

AgentHagu committed:
Add simple greeting and exit message

AgentHagu committed:
Add echo functionality

AgentHagu committed:
Update messages from chatbot

AgentHagu committed:
Add list functionality

AgentHagu committed:
Create Task.java class

AgentHagu committed:
Add mark functionalilty

AgentHagu committed:
Add unmark functionalilty

AgentHagu committed:
Create Todo, Deadline, Event class

AgentHagu committed:
Add functionality for different task types

AgentHagu committed:
Add input/output test file

AgentHagu committed:
Add error handling for invalid commands or inputs

AgentHagu committed:
Update tests for new error handling

AgentHagu committed:
Add delete functionality

AgentHagu committed:
Updated tests for delete support

AgentHagu committed:
Update error message for invalid command

AgentHagu committed:
Add basic functionality to load taskList from ./data/Neuro.txt

AgentHagu committed:
Add ability to create folder and file for save data if not found

AgentHagu committed:
Add feature to save after ending program

AgentHagu committed:
Merge branch 'branch-Level-7'

AgentHagu committed:
Catch NumberFormatException when calling mark or unmark

AgentHagu committed:
Update deadline to read and display different date time strings

AgentHagu committed:
Update tests for new features

AgentHagu committed:
Add more date time formatters for date time input validation

AgentHagu committed:
Merge branch 'branch-Level-8'

AgentHagu committed:
Create minimal classes for MoreOOP requirements

AgentHagu committed:
Refactor Neuro code to use more OOP with add tasks and bye functionality

AgentHagu committed:
Reimplement mark, unmark and delete commands with more OOP

AgentHagu committed:
Reimplement storage and autosave functionality

AgentHagu committed:
Handle corrupted save-file data

AgentHagu committed:
Clean up code and remove unwanted code

AgentHagu committed:
Merge remote-tracking branch 'origin/add-gradle-support'

AgentHagu committed:
Add minimal test cases using JUnit

AgentHagu committed:
Modify build.gradle to package App as neuro.jar

AgentHagu committed:
Add JavaDoc for most classes/methods in Neuro

AgentHagu committed:
Fix switch-case indentation

AgentHagu committed:
Add find command to Neuro Chatbot

AgentHagu committed:
Merge branch 'branch-A-JavaDoc'

AgentHagu committed:
Merge branch 'branch-A-CodingStandard'

AgentHagu committed:
Merge branch 'branch-Level-9'

AgentHagu committed:
Fix bugs and coding standard

AgentHagu committed:
Configure checkstyle for project

AgentHagu committed:
Rectify checkstyle issues and add more JavaDocs

AgentHagu committed:
Merge branch 'branch-A-CheckStyle'

AmadeusEka commented on src/main/java/neuro/Ui.java:
```diff
@@ -0,0 +1,117 @@
+                                                 @@%*+=======+++++====++#@@
+                                             @@#+==+=++=================+==+*@@
+                                     @@@%%@@*+=+===+=#*+=====================++*@@
+                       @    @ @@%####@*#@+==+====++@+===========================++@@
+                  @%@ %-=@@*-*%#####*@#========+=@++=+====++======================+=#@
+                 @*--%*--*@---@*###%#=========+%+==+#@#++#@%========================+=*@
+                @@%---#--=*---@*##@++=+======#*=++@+=+====+===========================+=#@
+                +-*#----------##%@#%++=====+%+==+@*+========================+============+@
+                @=-----=====---==--*#====++%===**%=====++=+=============+*%@%*=+==++=====++
+                 @*---======------%+=====+#=+=%=%+====+@+=#++=============+==+@#==+#=======
+                @*#+------------=%*%*===*#==#*=+#====#*@==@++=+================#++=@=======
+                ==#@#+======+#%%#===+#+##=##=--%=++=@=-@++@++====++====++========+=@++=====
+                #=======--===========%+###=---#++=%+=--#++@+========+==%@========+=@*+=====
+                %-===================*%=-----=#=#*=----+*+#*=======+@=#+@=========+%*+=====
+                %-===================++------%#%=-------#**#=++===+@#**=%=========*##+=====
+                @=====================#------*=---------#**%=====+##+#-=#=========%##======
+                @+====================@@@@@@*-=#+-------+%##*+=++@=%%=-+*=========@#*======
+                @#====================@@@@@@@@%=-%-------%%+#===%=+@=--*+========+*#+======
+                @%====================%*%@@@@@@@%=-------+@=#=+#*-*%---%=+=======#-%=======
+                @@====================#+====*@@@@@*-------@=+#=%=-#*--=%=======++#=#=======
+                %*+===================**======+@@#-----------%+#------=#@+=++=+=%=#++======
+                +=#====================#========%=-----------=@+---==+*#@+=#%++@==@========
+                +=%===================-#========-----------------#@@@@@@@%*=--===*+========
+                =+%+==================-#========--=+=----------+@@@@@@@@@@@@@@*-=%=========
+                =+**===================#+=====----%**#%%*=-----+*+===+*%@@@@@@@@%++========
+                ===#-==================**=-------##*******%#------========*@@@@@*+=========
+                =+*@====================#--------%**********-----============*@*=========+*
+                =+@@*==================-%--------%********@=----=============%+=========+=%
+                +@ @%===================*--------**======#=-----===========*%============%*
+                @   @===================+@=-------+====+%=-------=======+%#*@=+========++%*
+                    @*===================@-+@*=-------------------=========@++========++@**
+                   @%@===================#==+*-+%#+=-----------------=====%=+==========%***
+                   @*%+===================*==#*==+*=+###*==----------=*#%%============##***
+                  @#**#-=================-%===#*=-%#%*=*#===++++#@#****#*=+==========*#****
+                 @#***%+==================+*=*++*-*==*#====-=##+*#****%+=+%%+=======*%*****
+                @%*****%-=================-#-%+*%%+#%=-=#%#+=====+%*%@@%**%=======+##******
+                %******##==================+%@++=%@@@#%%%*=========%*****%======+=@********
+
+                """);
+    }
```
Amazing text art 🔥

AmadeusEka commented on src/main/java/neuro/task/Task.java:
```diff
@@ -0,0 +1,63 @@
+        this.isDone = false;
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as not done.
+     */
+    public void markUndone() {
+        this.isDone = false;
+    }
```
Perhaps the setter methods could be renamed to setDone for consistency?

AmadeusEka commented on src/main/java/neuro/task/TaskType.java:
```diff
@@ -0,0 +1,28 @@
+     * @param str the string representing the task type
+     * @return the corresponding {@code TaskType} enum value, or INVALID if the string is not recognized
+     */
+    public static TaskType stringToEnum(String str) {
+        return switch (str) {
+        case "todo" -> TODO;
+        case "deadline" -> DEADLINE;
+        case "event" -> EVENT;
+        default -> INVALID;
+        };
```
Great use of constants to increase readability of code!

AmadeusEka commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,181 @@
-| |_| | |_| |   <  __/
-|____/ \__,_|_|\_\___|
+    ___________________________________________________
+    Hii, I'm neuro.Neuro, your chatbot assistant!
```
Is the name neuro.Neuro intended?

AmadeusEka commented on src/main/java/neuro/command/FindCommand.java:
```diff
@@ -0,0 +1,37 @@
+        this.searchQuery = searchQuery;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        if (tasks.isEmpty()) {
+            ui.showMessage("You currently have no tasks.");
+        } else {
+            ui.showMessage("Here are the matching tasks in your list:");
+            int j = 1;
+            for (int i = 0; i < tasks.getSize(); i++) {
+                if (tasks.getTask(i).toString().contains(searchQuery)) {
+                    ui.showMessage(j + ". " + tasks.getTask(i));
+                    j++;
+                }
+            }
+        }
```
Perhaps you can include the JavaDoc for this execute method, similar to the other Command subclasses, to provide a more detailed description?

AmadeusEka commented on src/main/java/neuro/Ui.java:
```diff
@@ -0,0 +1,117 @@
+     *
+     * @param message the message to be shown
+     */
+    public void showMessage(String message) {
+        System.out.println("    " + message);
+    }
```
Perhaps the String.indent(n) method might be useful here?

AmadeusEka commented on src/main/java/neuro/command/DeleteCommand.java:
```diff
@@ -0,0 +1,49 @@
+        this.index = index;
+    }
+
+    /**
+     * Executes the delete command to delete the task at the index from the task list.
+     *
+     * @param tasks the task list on which the command operates
+     * @param ui the user interface for interacting with the user
+     * @param storage the storage for saving and loading tasks
+     */
+    @Override
```
Great descriptions in Javadoc that provide explanations and increases readibility and understanding of code throughout!

AmadeusEka reviewed (commented):
LDTM. Just some nits to fix

AgentHagu committed:
Add basic functionality for GUI, no proper Neuro logic incorporated yet

AgentHagu committed:
Add some Neuro logic into GUI

AgentHagu committed:
Change profile pictures and tweak UI slightly

AgentHagu committed:
Merge branch 'branch-Level-10'

AgentHagu committed:
Enable assertions in gradle run command

The code will now automatically run with assertions enabled, so that
we can now test certain assertions and values within the code that
we deem necessary in order for the code to function properly.

AgentHagu committed:
Add assertions into codebase

By using the "assert" feature, we are now documenting certain
assumptions within our codebase. For example, the program should
prevent users from creating tasks with empty descriptions. An
assertion was added for this to ensure that the parser prevents
this behaviour.

AgentHagu committed:
Refactor codebase

This is done to improve code quality of the code base. Some methods
were extracted to reduce the size of certain methods. Useless comments
were also removed and some needed code comments were added as well.

AgentHagu committed:
Change profile picture of user

AgentHagu committed:
Merge branch 'master' into branch-A-Assertions

AgentHagu committed:
Merge pull request #1 from AgentHagu/branch-A-Assertions

Add assertions into codebase

AgentHagu committed:
Merge pull request #2 from AgentHagu/branch-A-CodeQuality

Refactor codebase

AgentHagu committed:
Add tag command to Neuro

Users can now use the "tag" command to indicate:
1. The index of the task
2. The tag(s) to add to that task

E.g.
tag 1 fun
tag 2 graded URGENT

AgentHagu committed:
Merge pull request #3 from AgentHagu/C-Tagging

Add tag command to Neuro

AgentHagu committed:
Add .yml file and set up CI

AgentHagu committed:
Fix checkstyle issues and add JavaDocs

AgentHagu committed:
Improve UI by adding more colors and adding text boxes.

Mostly from the JavaFX tutorial provided by the course.

AgentHagu committed:
Fix minor issues

1. Add welcome message back to GUI version
2. Fix some indentations in replies
3. Delete redundant code

AgentHagu committed:
Merge pull request #5 from AgentHagu/branch-A-BetterGui

Improve UI by adding more colors and adding text boxes.

AgentHagu committed:
Fix minor issues

1. Update error messages for deadline to follow date/time format
2. Fix bug with assert and Find command
3. Fix bug with saving and loading tasks without any tags

AgentHagu committed:
Add Ui.png and update README.md

AgentHagu committed:
Add more test cases to ParserTest.java

AgentHagu committed:
Update NeuroTest.java with other tests besides dummy test

AgentHagu committed:
Change responses by Neuro to add a little bit more personality

AgentHagu committed:
Update tests for new personality and README.md for new release
